src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,257 @@
+    private static final int AGE_OF_TWENTY = 20;
+    private static final int AGE_OF_FORTY = 40;
+    private static final int NEGATIVE_AGE_OF_MINUS_FIVE = -5;
+    private static final int NEGATIVE_AGE_OF_MINUS_TWENTY = -20;
+    private static final int NEGATIVE_AGE_OF_MINUS_FIFTY = -50;
+    private static final String GOOD_LOGIN_FIRST = "login12345";
+    private static final String GOOD_LOGIN_SECOND = "admin12345";
+    private static final String GOOD_LOGIN_THIRD = "superAdmin12345";
+    private static final String GOOD_LOGIN_FORTH = "superlogin12345";
+    private static final String GOOD_LOGIN_FIFTH = "admin12345fff";
+    private static final String GOOD_LOGIN_SIXTH = "superAdmin12345SDVDFA";
+    private static final String GOOD_LOGIN_SEVENTH = "login12345sdsfvs";
+    private static final String GOOD_LOGIN_EIGHT = "admin123457e7ch";
+    private static final String GOOD_LOGIN_NINTH = "superAdmin12345asadvc";
+    private static final String GOOD_LOGIN_TENTH = "superAdmin12345asadvcDSSDCADS";
+    private static final String GOOD_PASSWORD_FIRST = "password12345";
+    private static final String GOOD_PASSWORD_SECOND = "password3888112";
+    private static final String GOOD_PASSWORD_THIRD = "passwordMkH61345555";
+    private static final String EMPTY_LOGIN_OR_PASSWORD = "";
+    private static final String LOGIN_LENGTH_OF_TWO = "1h";
+    private static final String LOGIN_LENGTH_OF_FOUR = "ABCD";
+    private static final String PASSWORD_LENGTH_OF_THREE = "j2k";
+    private static final String PASSWORD_LENGTH_OF_FIVE = "KHw7R";
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        User user = new User();
+        user.setLogin(GOOD_LOGIN_FIRST);
+        user.setPassword(GOOD_PASSWORD_FIRST);
+        user.setAge(AGE_OF_TWENTY);
+        Storage.people.add(user);
+        for (int i = 0; i < 5; i++) {
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_FIRST);
+        user2.setLogin(GOOD_LOGIN_SECOND);
+        user3.setLogin(GOOD_LOGIN_THIRD);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(null);
+        user2.setAge(null);
+        user3.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
```

Comment:
The test method `register_nullAge_notOk` could be simplified by using a single User object and changing its age to null, rather than creating multiple User objects with the same data. This would make the test more concise and easier to understand.